app/src/main/java/kth/jjve/xfran/EventPlanActivity.java

```diff
@@ -81,29 +91,79 @@ protected void onCreate(Bundle savedInstanceState) {
             s_eventName = mWorkout.getTitle();
             eventNameTV.setText(s_eventName);
         }
+
+        /*------ DATE PICKER DIALOG -----*/
+        eventDateInput.setOnClickListener(v -> {
+            // calender class's instance and get current date , month and year from calender
+            final Calendar calendar = Calendar.getInstance();
+            int mYear = calendar.get(Calendar.YEAR);
+            int mMonth = calendar.get(Calendar.MONTH); // ADD +1 to get actual month!!!
+            int mDay = calendar.get(Calendar.DAY_OF_MONTH);
+
+            datePickerDialog = new DatePickerDialog(EventPlanActivity.this,
+                    (view, year, month, day) -> {
+                        // set day of month , month and year value in the edit text
+                        String date = day + "/"  + (month + 1) + "/" + year;
+                        eventDateInput.setText(date);
+                    }, mYear, mMonth, mDay);
+            datePickerDialog.getDatePicker().setFirstDayOfWeek(Calendar.MONDAY);
+            datePickerDialog.show();
+        });
+
+        /*------ TIME PICKER DIALOG -----*/
+        // TODO: see if can be implemented in method?
+        eventStartTimeInput.setOnClickListener(v -> {
+            Calendar mCurrentTime = Calendar.getInstance();
```

Comment:
line 116-124 is the same as line 128-136 which means it could be made into a method, which also makes the code more readable.